syntax = "proto3";

package communication;

message User {
    int32 id = 1;
    string name = 2;
    string email = 3;
    repeated string roles = 4;
}

message UserRequest {
    int32 user_id = 1;
}

message UserResponse {
    User user = 1;
    bool success = 2;
    string error_message = 3;
}

service UserService {
    rpc GetUser (UserRequest) returns (UserResponse);
}